lib/src/stream_extensions.dart

```diff
@@ -54,4 +54,17 @@ extension StreamExtensions<T> on Stream<T> {
     });
     return completer.future;
   }
+
+  /// Returns a view of this stream that eagerly buffers all events until
+  /// `listen()` is called.
+  Stream<T> buffer() {
+    late StreamSubscription<T> subscription;
+    var controller = StreamController<T>(
+        onPause: () => subscription.pause(),
+        onResume: () => subscription.resume(),
+        onCancel: () => subscription.cancel());
+    subscription = listen(controller.add,
```

Comment:
I think it is safe to pass `cancelOnError: false` here.

If the result is listened with `cancelOnError: false`, we'd need it here too or that parameter won't work.

If the result is listened with `cancelOnError: true`, it should be OK. We'll add multiple errors here, but close the subscription once the result's listener closes on the first error.

Reply:
I agree—but isn't `cancelOnError: false` the default?

Reply:
🤦 - it is the default.